pkg/mysql/server.go

```diff
@@ -555,43 +553,33 @@ func (l *Listener) ExecuteCommand(c *Conn, ctx *proto.Context) error {
 			c.recycleReadPacket()
 			result, warn, err := l.executor.ExecutorComQuery(ctx)
 			if err != nil {
-				if werr := c.writeErrorPacketFromError(err); werr != nil {
-					log.Error("Error writing query error to client %v: %v", l.connectionID, werr)
-					return werr
+				if wErr := c.writeErrorPacketFromError(err); wErr != nil {
+					log.Error("Error writing query error to client %v: %v", l.connectionID, wErr)
+					return wErr
 				}
 				return nil
 			}
-			if len(result.GetFields()) == 0 {
+			if rlt := result.(*Result); len(rlt.Fields) == 0 {
```

Comment:
Keep using interface function, maybe there are other result implementations in the future.